skopt/optimizer/optimizer.py

```diff
@@ -446,6 +450,31 @@ def tell(self, x, y, fit=True):
                 warnings.simplefilter("ignore")
                 est.fit(self.space.transform(self.Xi), self.yi)
 
+            # if we are using noisy expected improvement generate N variants
+            # of the observed data and compute a GP for each variant.
+            # The variable names reflect the notation used in the original 
+            # paper (Algorithm 1)
+            if self.acq_func == "noisyEI":
+                mu, sigma = est.predict(self.Xi, return_cov=True)
+                A = np.linalg.cholesky(sigma + np.eye(len(self.Xi)) * 1e-10)
+                # skip the first sobol (all zeros) or we obtain inf and nans
+                t = sobol_sample(self.noisyEI_N_variants+1, len(self.Xi))[1:]
+                yis = norm.ppf(t).dot(A.T) + mu
+
+                noisy_estimators = list()
+                for variant in range(self.noisyEI_N_variants):
+                    # fit a new estimator
+                    noisy_est = clone(self.base_estimator_)
+                    with warnings.catch_warnings():
+                        warnings.simplefilter("ignore")
+                        noisy_est.fit(self.space.transform(self.Xi),
+                                      yis[variant])
+                    noisy_estimators.append(noisy_est)
+
+                # use this list as estimator, it will be handled by the
+                # acquisition function
+                est = noisy_estimators
```

Comment:
This has the disadvantage that the result object returned after optimization only containts the sampled noiseless GPs and not the original noisy GP.
In my opinion result.models should always only contain the original GPs. Maybe we need to introduce a new variable for the variants.